src/Microsoft.ComponentDetection.Detectors/nuget/NuGetProjectModelProjectCentricComponentDetector.cs

```diff
@@ -264,19 +103,27 @@ private void NavigateAndRegister(
         ISingleFileComponentRecorder singleFileComponentRecorder,
         LockFileTargetLibrary library,
         string parentComponentId,
-        HashSet<string> dotnetRuntimePackageNames,
+        FrameworkPackages frameworkPackages,
         HashSet<string> visited = null)
     {
-        if (this.IsAFrameworkComponent(dotnetRuntimePackageNames, library.Name, library.Dependencies)
-            || library.Type == ProjectDependencyType)
+        if (library.Type == ProjectDependencyType)
         {
             return;
         }
 
+        var isFrameworkComponent = frameworkPackages.IsAFrameworkComponent(library.Name, library.Version);
+        var isDevelopmentDependency = IsADevelopmentDependency(library);
+
         visited ??= [];
 
         var libraryComponent = new DetectedComponent(new NuGetComponent(library.Name, library.Version.ToNormalizedString()));
```

Comment:
If we happen to be dropping the framework version due to dependency resolution, should this version be higher? Namely folks using `System.Text.Json 6.0.0` but actually running a dotnet core app with `System.Text.Json 7.0.4`, shouldn't the framework version win here, or which one ends up getting used by the project?

Reply:
When the framework wins it means the version isn't shipping with the app (for framework dependent app) or that the framework's runtime pack is winning (for self-contained).  In the first case there is no need for an alert - nothing for the app to do.  In the latter case, the runtime package will be seen by CG and the CVE will apply to the framework runtime package.